src/dftd4/param.f90

```diff
@@ -45,18 +45,196 @@ module dftd4_param
          & p_bmk, p_b1b95, p_pwb6k, p_otpss, p_ssb, p_revssb, &
          & p_pbesol, p_hse06, p_pbexalpha, p_pbehpbe, p_hcth407, &
          & p_n12, p_pkzb, p_thcth, p_m11l, p_mn15l, p_mpwb1k, &
-         & p_mpw1kcis, p_mpwkcis1k, p_pbeh1pbe, p_pbe1kcis, p_b97_1, &
-         & p_b97_2, p_b98, p_hiss, p_hse03, p_revtpssh, p_tpss1kcis, &
+         & p_mpw1kcis, p_mpwkcis1k, p_mpw1b95, p_pbeh1pbe, p_pbe1kcis, &
+         & p_b97_1, p_b97_2, p_b98, p_hiss, p_hse03, p_revtpssh, p_tpss1kcis, &
          & p_m05, p_m052x, p_m08hx, p_lcwhpbe, p_mn12l, p_tauhcthhyb, &
-         & p_sogga11x, p_n12sx, p_mn12sx, p_mn15, p_glyp, p_bop, &
-         & p_mpw1b95, p_revpbe0dh, p_revtpss0, p_revdsdpbep86, p_revdsdpbe, &
+         & p_sogga11x, p_n12sx, p_mn12sx, p_mn15, p_glyp, &
+         & p_revpbe0dh, p_revtpss0, p_revdsdpbep86, p_revdsdpbe, &
          & p_revdsdblyp, p_revdodpbep86, p_am05, p_hse12, p_hse12s, &
-         & p_r2scanh, p_r2scan0, p_r2scan50
+         & p_r2scanh, p_r2scan0, p_r2scan50, p_last
    end enum
    integer, parameter :: df_enum = kind(p_invalid)
 
+
+   ! Group different spellings/names of functionals
+   type functional_group
+       character(len=:), allocatable :: names(:)
+   contains
+       procedure :: set_names
+   end type functional_group
+
 contains
 
+subroutine set_names(self, input_names)
+   class(functional_group), intent(inout) :: self
```

Comment:
This could be safely using type instead of class and out instead of inout for the intent.